workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java

```diff
@@ -139,7 +140,22 @@ private Payload handleAction(KitchenSink.Action action) {
       throw ApplicationFailure.newFailure(error.getFailure().getMessage(), "");
     } else if (action.hasContinueAsNew()) {
       KitchenSink.ContinueAsNewAction continueAsNew = action.getContinueAsNew();
-      Workflow.continueAsNew(continueAsNew.getArgumentsList().get(0));
+
+      // Preserve search attributes across continue-as-new
+      io.temporal.common.SearchAttributes currentSearchAttributes =
+          Workflow.getTypedSearchAttributes();
+
+      if (currentSearchAttributes != null && currentSearchAttributes.size() > 0) {
+        // Explicitly preserve search attributes
+        ContinueAsNewOptions options =
+            ContinueAsNewOptions.newBuilder()
+                .setTypedSearchAttributes(currentSearchAttributes)
+                .build();
+        Workflow.continueAsNew(options, continueAsNew.getArgumentsList().get(0));
+      } else {
+        // No search attributes to preserve
+        Workflow.continueAsNew(continueAsNew.getArgumentsList().get(0));
```

Comment:
Nit: I think in principle you could just always write this like the first  branch and it'd have the same effect either way.

AI comments inside the branches not super useful 😄

Reply:
lol these comments were mine, rip 🙃 
I have grown indistinguishable from AI